src/legacy/ui/public/agg_types/buckets/filters.js

```diff
@@ -40,32 +44,36 @@ export const filtersBucketAgg = new BucketAggType({
     {
       name: 'filters',
       editorComponent: FiltersParamEditor,
-      default: [ { input: {}, label: '' } ],
+      default: [ { input: { query: '', language: config.get('search:queryLanguage') }, label: '' } ],
       write: function (aggConfig, output) {
         const inFilters = aggConfig.params.filters;
         if (!_.size(inFilters)) return;
 
+        inFilters.forEach((filter) => {
+          const persistedLog = getQueryLog('filtersAgg', filter.input.language);
+          persistedLog.add(filter.input.query);
+        });
+
         const outFilters = _.transform(inFilters, function (filters, filter) {
-          const input = _.cloneDeep(filter.input);
+          let input = _.cloneDeep(filter.input);
 
           if (!input) {
             console.log('malformed filter agg params, missing "input" query'); // eslint-disable-line no-console
             return;
           }
 
-          const query = input.query = luceneStringToDsl(input.query);
+          const query = input = buildEsQuery(aggConfig.getIndexPattern(), [input], [], config);
+
           if (!query) {
             console.log('malformed filter agg params, missing "query" on input'); // eslint-disable-line no-console
             return;
           }
-          const config = chrome.getUiSettingsClient();
-          const queryStringOptions = config.get('query:queryString:options');
-
-          decorateQuery(query, queryStringOptions);
 
-          const matchAllLabel = (filter.input.query === '' && _.has(query, 'match_all')) ? '*' : '';
-          const label = filter.label || matchAllLabel || _.get(query, 'query_string.query') || angular.toJson(query);
-          filters[label] = input;
+          const matchAllLabel = filter.input.query === '' ? '*' : '';
+          const label = filter.label
+            || matchAllLabel
+            || (typeof filter.input.query === 'string' ? filter.input.query : angular.toJson(filter.input.query));
```

Comment:
With the only change in the filter query being the addition of the query language that changes the shape of a filter from `filters: [
                    {
                      input: {
                        query: 'response:200',
                      },
                      label: '',
                    },`
to `{
                        input: {
                          query: 'response:200',
                          language: 'lucene',
                        },
                        label: '',
                      }`
will the filter.input.query not always be a string?

Reply:
Users can also enter full query DSL instead of a lucene query string, in which case `filter.input.query` will be an object.

For example, you could type in `{"exists": {"field": "phpbytes"}}`. We'll detect and parse this as an object.